README.md

```diff
@@ -12,3 +12,22 @@ aims to solve this problem.
 This essentially gives us rebase-like control over exactly how the final
 branch will look while at the same time making it as safe as branching and
 merging without rewriting history: the best of both worlds.
+
+## How to use it
+
+The `shell` command in the top of this repository is the entry point for using
+the code in this repository. It spawns a sub-shell with `$PATH` adjusted
+appropriately to activate it.
+
+You also need a git repository to work with. When you run `shell` it will look
+to see if your current working directory is part of one. If it is, it will
+install necessary hooks into the repository to create an obsolescence graph of
+your changes. If it isn't part of an existing git repository, it will create one
+in a temporary directory and populate it with some dummy files that you can play
```

Comment:
It doesn't do this yet. I wonder if I should take this out of the README until it is actually needed.